src/Synthesiser.cpp

```diff
@@ -747,17 +750,18 @@ void Synthesiser::emitCode(std::ostream& out, const RamStatement& stmt) {
             auto ctxName = "READ_OP_CONTEXT(" + synthesiser.getOpContextName(rel) + ")";
             auto identifier = aggregate.getIdentifier();
 
-            // get the tuple type working with
+            // aggregate tuple storing the result of aggregate
             std::string tuple_type = "ram::Tuple<RamDomain," + toString(std::max(1u, arity)) + ">";
 
             // declare environment variable
-            out << tuple_type << " env" << identifier << ";\n";
+            out << "ram::Tuple<RamDomain,1> env" << identifier << ";\n";
```

Comment:
If this change is valid, then presumably the definition of `key` on line 814 can also be modified, and the definition of the string tuple_type removed

Reply:
No - that is not correct. We have two different env variables. One for the aggregate's result the other one for the aggregate loop. The key belongs to the env of the aggregate loops.

Reply:
I wasn't suggesting combining the two. At the moment key is defined by `tuple_type` as being the greater of size 1 or arity, then filled by printKeyTuple with a size of arity. This was because previously `env` was also specified by tuple_type and needed to be a minimum of size 1. Now that env is specified to be of size 1, `key` can be specified to be of size arity - `Ram::Tuple<RamDomain, arity> key`. With that change, `tuple_type` is not needed at all.

Reply:
Ah - I see. Yes indeed.